library/src/main/java/jp/co/cyberagent/android/gpuimage/filter/GPUImageFilterGroup.kt

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (C) 2018 CyberAgent, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jp.co.cyberagent.android.gpuimage.filter
+
+import android.annotation.SuppressLint
+import android.opengl.GLES20
+import jp.co.cyberagent.android.gpuimage.GPUImageRenderer
+import jp.co.cyberagent.android.gpuimage.util.Rotation
+import jp.co.cyberagent.android.gpuimage.util.TextureRotationUtil
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.nio.FloatBuffer
+
+/**
+ * Resembles a filter that consists of multiple filters applied after each
+ * other.
+ */
+open class GPUImageFilterGroup @JvmOverloads constructor(private val filters: MutableList<GPUImageFilter> = mutableListOf()) :
+    GPUImageFilter() {
+
+    private lateinit var frameBufferIndexMap: Map<Int, Int>
+    private var mergedFilters: MutableList<GPUImageFilter>? = null
+    private var frameBuffers: IntArray? = null
+    private var frameBufferTextures: IntArray? = null
+    private val glCubeBuffer: FloatBuffer
+    private val glTextureBuffer: FloatBuffer
+    private val glTextureFlipBuffer: FloatBuffer
+    /**
+     * Instantiates a new GPUImageFilterGroup with the given filters.
+     *
+     * @param filters the filters which represent this filter
+     */
+    /**
+     * Instantiates a new GPUImageFilterGroup with no filters.
+     */
+    init {
+        updateMergedFilters()
+        glCubeBuffer = ByteBuffer.allocateDirect(GPUImageRenderer.CUBE.size * 4)
+            .order(ByteOrder.nativeOrder())
+            .asFloatBuffer()
```

Comment:
'''suggestion
// Allocate a direct byte buffer for the cube vertices:
glCubeBuffer = ByteBuffer.allocateDirect(GPUImageRenderer.CUBE.size * 4)
.order(ByteOrder.nativeOrder())
.asFloatBuffer()